installer/lib/installer.py

```diff
@@ -431,7 +370,43 @@ def set_sys_path(venv_path: Path) -> None:
     sys.path.append(str(Path(venv_path, lib, "site-packages").expanduser().resolve()))
 
 
-def get_torch_source() -> (Union[str, None], str):
+def get_github_releases() -> tuple[list, list] | None:
```

Comment:
I'm not confident in this as it assumes the pypi releases are exactly the same as github tags. That's _probably_ true, but as you saw the sorting is a bit off.

You can get the releases from pypi and and sort/group according to version number and release type: https://gist.github.com/psychedelicious/d589c53284f88cad3ea57f320b21687

Reply:
pypi releases must always be exactly the same as github tags. anything else would indicate a serious problem in our release process.
Using github as the source of truth for pre-release status is preferable because we can mark any version in Github as a pre-release (and unmark if needed), as opposed to having to parse version strings received from PyPi. also, the JSON payload returned by github is significantly smaller and contains more relevant information.
if this is still a concern in the future, we can revisit.

Reply:
Thanks for explaining, makes sense